roles/include_components/README.md

```diff
@@ -1,15 +1,46 @@
-# include_components role
+# redhat.ocp.include_components role - create and attach DCI components to the DCI jobs
 
-The `include_components` role adds components in the DCI jobs from software resources available in the ansible controller node, for example: rpm packages, and git repositories used during the job execution.
+- [Synopsis](https://galaxy.ansible.com/ui/repo/published/redhatci/ocp/content/role/include_components/README.md#synopsis)
+- [Parameters](https://galaxy.ansible.com/ui/repo/published/redhatci/ocp/content/role/include_components/README.md#parameters)
+- [Examples](https://galaxy.ansible.com/ui/repo/published/redhatci/ocp/content/role/include_components/README.md#examples)
+
+# Synopsis
+
+The `include_components` role creates and attaches DCI components to the DCI jobs by generating them from provided resources. Possible resource inputs include RPM packages, Git repositories, and Podman images.
 
 Follow this post to know more about [components](https://blog.distributed-ci.io/automate-dci-components.html).
 
-## Configuration
+# Parameters
 
 Variables used by this role:
 
 | Setting        | Required | Type   | Description                                                        |
 | -------------- | -------- | ------ | -------------------------------------------------------------------|
-| ic\_rpms       | True     | List   | List of RPM names to include as components                         |
-| ic\_gits       | True     | List   | List of directories from GIT repositories to include as components |
-| ic\_dev\_gits  | True     | List   | List of complimentary directories from GIT repositories to include as components |
+| ic_rpms        | False    | List   | Optional. List of RPM names to include as components                         |
+| ic_gits        | True     | List   | Mandatory. List of directories from GIT repositories to include as components |
+| ic_dev_gits    | True     | List   | Mandatory. List of complimentary directories from GIT repositories to include as components |
+| ic_repo_commits | False    | List | Optional. List of URLs pointing directly to code commits `[https://hosting.com/organisation/reponame/commit/b6bcc3506c0d84baa0c020f6b776a181b931f57a, https://github.com/redhat-openshift-ecosystem/openshift-preflight/commit/b6bcc3506c0d84baa0c020f6b776asfasfasdfa]` to include as components |
```

Comment:
Make a reference to the example below to avoid the extra typing. And include the expected formatting, like the one used in the assertion

Reply:
Sure, I've added the expected formatting. However, I kept the examples, as they are typically included in the Ansible documentation